### ISSUE TITLE
gltf_viewer sample crashes if you try to use DoF with Metal

### ISSUE THREAD

**Issue by jwwalker** [google/filament#3733]
**Describe the bug**
The `gltf_viewer` sample crashes if you try to turn on DoF when using a Metal backend.

**To Reproduce**
1. On macOS, run `gltf_viewer -a metal` from the Terminal.
2. Open the disclosure triangle for Camera settings.
3. Click the checkbox to the left of DoF

**Expected behavior**
Nothing much immediately happens.

**Actual behavior**
The sample crashes, and logs:

```
-[MTLTextureDescriptorInternal validateWithDevice:]:1248: failed assertion `Texture Descriptor Validation
Texture swizzling is not compatable with MTLTextureUsageRenderTarget'
```

**Desktop:**
 - OS: macOS 11.3 beta
 - GPU: Radeon Pro 560X
 - Backend: Metal

**Additional context**
As far as I know, this is the only sample that does DoF at all.

**Comment by romainguy** [comment]
This is a problem we were discussing recently. Swizzling has minium requirements on macOS/iOS. @bejado would have all the details.

**Comment by bejado** [comment]
This will be fixed by #3732
